course5/homework05/vector2d.h

```diff
@@ -0,0 +1,47 @@
+#pragma once
+#include <iostream>
+#include <string>
+
+namespace vector
+{
+class Vector2D
+{
+public:
+    enum class System { CARTESIAN, POLAR };
+
+    Vector2D(double initX = 0.0, double initY = 0.0);
+    Vector2D(double arg1, double arg2, System system);
```

Comment:
Interesting API design, but flawed: it requires a runtime check inside of the function.
Google: "named constructor idiom". This could be a nice solution here.